tests/tracer/test_correlation_log_context.py

```diff
@@ -0,0 +1,136 @@
+import pytest
+import structlog
+
+from ddtrace import Tracer
+from ddtrace import config
+from ddtrace import tracer
+from ddtrace.opentracer.tracer import Tracer as OT_Tracer
+from tests.utils import override_global_config
+
+
+@pytest.fixture
+def global_config():
+    config.service = "test-service"
+    config.env = "test-env"
+    config.version = "test-version"
+    yield
+    config.service = config.env = config.version = None
+
+
+def tracer_injection(logger, log_method, event_dict):
+    """Inject tracer information into custom structlog log"""
+    correlation_log_context = tracer.get_log_correlation_context()
+
+    # add ids and configs to structlog event dictionary
+    event_dict["dd_correlation_context"] = correlation_log_context
```

Comment:
ah, we'll still want `event_dict["dd.trace_id"]`

we also need to be sure a `DDLogRecord` can be JSON encoded.

Reply:
Just to clarify, when you suggested `event_dict.update(correlation_log_context)` did you mean to store the actual `DDLogRecord` in the event dictionary? Since the `DDLogRecord` object isn't a dictionary that can be used to update.

Reply:
yeah, the end result needs to be the same (e.g. `event_dict["dd.trace_id"]` is the only way the correlation with the logs product will work, or `event_dict["dd"]["trace_id"]`), but I am wondering if we can make people's lives easier, instead of them needing to manually add each element to the `event_dict`


or what if they use another library that works a little differently, or.. something? just wondering if we want/need the flexibility here?

for the same reason, it might be nice if `DDLogRecord` was JSON serializable.


what do you think?